### PR TITLE
chore: bumpenvs 606e7f2

### PR DESCRIPTION
**Pull request by drh-determined-ai**
## Description

Bump environment images based on `environments` commit 606e7f22c75250e8c3a5e425ea0f1a3930219abc "TF2_VERSION 2.12".
Changes to the example gbt_titanic_estimator have been held back because Keras estimates do not exist for tensorflow>=2.11.0. This example will be updated to use decision forests in [MLG-442](https://determinedai.atlassian.net/browse/MLG-442).

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[MLG-442]: https://hpe-aiatscale.atlassian.net/browse/MLG-442?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mpkouznetsov — COMMENTED** [review]
A couple updates would be useful. Plus, this needs a release note saying something like

```
:orphan:

**Improvements**

-  Images: Update default environment images to Tensorflow 2.12 and Horovod 0.27.0.
```

**Comment by drh-determined-ai** [comment]
This PR is being closed because we cannot upgrade to TensorFlow 2.12 at this time (see the comments in [MLG-358](https://hpe-aiatscale.atlassian.net/browse/MLG-358)).
